marathon_lb.py

```diff
@@ -166,27 +166,35 @@ def __init__(self, hosts, health_check, strict_mode, auth, ca_cert=None):
             self.__verify = ca_cert
 
     def api_req_raw(self, method, path, auth, body=None, **kwargs):
+        response, exception = None, None
         for host in self.__hosts:
             path_str = os.path.join(host, 'v2')
 
             for path_elem in path:
                 path_str = path_str + "/" + path_elem
+            try:
+                response = requests.request(
+                    method,
+                    path_str,
+                    auth=auth,
+                    headers={
+                        'Accept': 'application/json',
+                        'Content-Type': 'application/json'
+                    },
+                    timeout=(3.05, 46),
+                    **kwargs
+                )
 
-            response = requests.request(
-                method,
-                path_str,
-                auth=auth,
-                headers={
-                    'Accept': 'application/json',
-                    'Content-Type': 'application/json'
-                },
-                timeout=(3.05, 46),
-                **kwargs
-            )
+                logger.debug("%s %s", method, response.url)
+                if response.status_code == 200:
+                    break
+            except Exception as e:
```

Comment:
Can we catch a smaller set of exceptions? Perhaps just `requests.exceptions.Timeout`?

Reply:
Pretty much everyone listed at https://docs.python.org/3/library/exceptions.html except `IOError`, as that's the basis for socket.error. Most everything else isn't specific to a host, e.g  `IndexError` or `AssertionError `. Those will be raised if there are issues in the code that gets run.